src/js/Edit/L.PM.Edit.CircleMarker.js

```diff
@@ -104,18 +105,17 @@ Edit.CircleMarker = Edit.extend({
     // disable dragging, as this could have been active even without being enabled
     this.disableLayerDrag();
 
+    this._layer.off('contextmenu', this._removeMarker, this);
+
     // only fire events if it was enabled before
     if (!this.enabled()) {
```

Comment:
@codeofsumit 
`CircleMarker` is the only Shape which checks if it was enabled to fire events. Should we remove this or add this to the other Shapes too?

Reply:
remove it for now and lets see if it creates problems. I guess there was a reason for this but my comment should have stated it